src/osal/lv_freertos.c

```diff
@@ -193,17 +188,30 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 {
     lv_result_t lvRes = LV_RESULT_OK;
 
+    /* If the cond is uninitialized, perform initialization. */
+    prvCheckCondInit(pxCond);
+
 #if USE_FREERTOS_TASK_NOTIFY
-    LV_UNUSED(pxCond);
+    TaskHandle_t current_task_handle = xTaskGetCurrentTaskHandle();
 
-    /* Wait for other task to notify this task. */
-    ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+    _enter_critical();
+    BaseType_t signal_sent = pxCond->xSyncSignal;
+    pxCond->xSyncSignal = pdFALSE;
+    if(signal_sent == pdFALSE) {
+        /* The signal hasn't been sent yet. Tell the sender to notify this task */
+        pxCond->xTaskToNotify = current_task_handle;
+    }
+    /* If we have a signal from the other task, we should not ask to be notified */
+    _exit_critical();
+
```

Comment:
After exiting from critical section the `pxCond->xSyncSignal` might get changed, so this is not thread safe.